swift-sdk/Internal/ApiClient.swift

```diff
@@ -186,55 +230,17 @@ class ApiClient: ApiClientProtocol {
     func inAppConsume(inAppMessageContext: InAppMessageContext, source: InAppDeleteSource?) -> Future<SendRequestValue, SendRequestError> {
         return send(iterableRequestResult: createRequestCreator().createTrackInAppConsumeRequest(inAppMessageContext: inAppMessageContext, source: source))
     }
-    
-    func disableDevice(forAllUsers allUsers: Bool, hexToken: String) -> Future<SendRequestValue, SendRequestError> {
-        return send(iterableRequestResult: createRequestCreator().createDisableDeviceRequest(forAllUsers: allUsers, hexToken: hexToken))
-    }
-    
-    func convertToURLRequest(iterableRequest: IterableRequest) -> URLRequest? {
-        switch iterableRequest {
-        case let .get(getRequest):
-            return IterableRequestUtil.createGetRequest(forApiEndPoint: endPoint, path: getRequest.path, headers: createIterableHeaders(), args: getRequest.args)
-        case let .post(postRequest):
-            return IterableRequestUtil.createPostRequest(forApiEndPoint: endPoint, path: postRequest.path, headers: createIterableHeaders(), args: postRequest.args, body: postRequest.body)
-        }
-    }
-    
-    func send(iterableRequestResult result: Result<IterableRequest, IterableError>) -> Future<SendRequestValue, SendRequestError> {
-        switch result {
-        case let .success(iterableRequest):
-            return send(iterableRequest: iterableRequest)
-        case let .failure(iterableError):
-            return SendRequestError.createErroredFuture(reason: iterableError.localizedDescription)
-        }
-    }
-    
-    func send(iterableRequest: IterableRequest) -> Future<SendRequestValue, SendRequestError> {
-        guard let urlRequest = convertToURLRequest(iterableRequest: iterableRequest) else {
-            return SendRequestError.createErroredFuture()
-        }
-        
-        return NetworkHelper.sendRequest(urlRequest, usingSession: networkSession)
-    }
-    
-    private func createRequestCreator() -> RequestCreator {
-        guard let authProvider = authProvider else {
-            fatalError("authProvider is missing")
-        }
-        
-        return RequestCreator(apiKey: apiKey, auth: authProvider.auth, deviceMetadata: deviceMetadata)
+}
+
+// DEPRECATED
+extension ApiClient {
```

Comment:
```suggestion
extension ApiClient: ApiClientProtocol {
```
This way the intention is clear.

Reply:
Looks like this isn't possible, I get "Redundant conformance of 'ApiClient' to protocol 'ApiClientProtocol'", otherwise I would. That's a new one.